### PR TITLE
Add tomlc99 library

### PR DESCRIPTION
**Pull request by metalefty**
This is a preparation for getting started TOML config parser.

 * Add tomlc99 as submodule
   * and integrate it with xrdp build
 * Add option to show third-party licenses

tomlc99 library will be installed the following location:
```
./usr/local/lib/xrdp/libxrdpapi.so.0.0.0
./usr/local/lib/xrdp/libxrdpapi.so
./usr/local/lib/xrdp/libxup.la
./usr/local/lib/xrdp/libxrdp.so
./usr/local/lib/xrdp/libcommon.so.0
./usr/local/lib/xrdp/libmc.la
./usr/local/lib/xrdp/libipm.so.0
./usr/local/lib/xrdp/libmc.a
./usr/local/lib/xrdp/libxrdp.so.0
./usr/local/lib/xrdp/libipm.so
./usr/local/lib/xrdp/libipm.la
./usr/local/lib/xrdp/libxrdp.a
./usr/local/lib/xrdp/libxrdp.la
./usr/local/lib/xrdp/libmc.so
./usr/local/lib/xrdp/libxup.a
./usr/local/lib/xrdp/libvnc.a
./usr/local/lib/xrdp/libxup.so
./usr/local/lib/xrdp/libxrdpapi.a
./usr/local/lib/xrdp/libxrdpapi.la
./usr/local/lib/xrdp/libipm.so.0.0.0
./usr/local/lib/xrdp/libvnc.la
./usr/local/lib/xrdp/libipm.a
./usr/local/lib/xrdp/libtoml.a
./usr/local/lib/xrdp/libcommon.so.0.0.0
./usr/local/lib/xrdp/libcommon.so
./usr/local/lib/xrdp/libxrdp.so.0.0.0
./usr/local/lib/xrdp/libxrdpapi.so.0
./usr/local/lib/xrdp/libtoml.la
./usr/local/lib/xrdp/libtoml.so.1.0.0
./usr/local/lib/xrdp/libtoml.so
./usr/local/lib/xrdp/libcommon.la
./usr/local/lib/xrdp/libvnc.so
./usr/local/lib/xrdp/libcommon.a
./usr/local/lib/xrdp/libtoml.so.1
```

### CODE REVIEW

**Comment by matt335672** [comment]
I like the way you've addressed the licences first - that's very important

I'd also suggest the following:-

1) If we're using g++ as a C compiler (for CI), build the third party stuff with gcc instead.
2) Skip running `astyle` in the third_party directory, as we have no control over what goes on in there.

**Comment by metalefty** [comment]
You probably meant cppcheck, not astyle. I'll try to build tomlc99 with GCC for g++ case later, thanks!

**Comment by matt335672** [comment]
Yes I did mean cppcheck!

**Comment by metalefty** [comment]
Switched from submodule to subtree. I think subtree is more suitable if we make additional changes to the original source. Additional changes means replacing Makefile here.

File: scripts/run_astyle.sh [hunk 1]
```diff
@@ -65,7 +65,7 @@ if [ ! -f "astyle_config.as" ]; then
     exit 2
 fi
 
-ASTYLE_FLAGS="--options=astyle_config.as ./\*.c ./\*.h"
+ASTYLE_FLAGS="--options=astyle_config.as --exclude=third_party/tomlc99 ./\*.c ./\*.h"
```
Comment (matt335672) [inline scripts/run_astyle.sh]: Probably don't need the "/tomlc99" - we won't want to check any third-party libs.

**Review by matt335672 — APPROVED** [review]
I'm happy with that - the subtree works a lot better for this. The following two commands helped me to see what was going on:-

```
git remote add cktan_tomlc99 https://github.com/cktan/tomlc99.git
git diff HEAD:third_party/tomlc99 cktan_tomlc99/master
```

There doesn't appear to be any version embedded in tomlc99 which is a pity. It might make it harder to work out if we need updates.

File: scripts/run_astyle.sh (hunk 1 above)
Comment (metalefty) [reply scripts/run_astyle.sh]: Indeed.

**Comment by metalefty** [comment]
I have created a blank wiki page [Migration to TOML](https://github.com/neutrinolabs/xrdp/wiki/Migration-to-TOML). 

I intend to separate TOML stuff not put everything on v1.0 release note.

**Comment by matt335672** [comment]
The wiki page is a good idea. There's quite a bit of design work to do on the new files, I think.

**Pull request merged**